tests/fe/p1nc_01.cc

```diff
@@ -0,0 +1,77 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2015 - 2016 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+
+// check the P1NC element on a square
+
+#include "../tests.h"
+#include <deal.II/base/logstream.h>
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/grid/tria.h>
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/fe/fe_values.h>
+
+#include <deal.II/fe/fe_p1nc.h>
+
+#include <fstream>
+#include <string>
+
+#define PRECISION 5
```

Comment:
This is ```#define PRECISION 5``` The compiler will replace ```PRECISION ``` by 5 everywhere but the problem it's that it works like a global variable with a constant value. There is no problem with that in a test but it just makes the code harder to read for no reason. You could use a normal variable.